<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1412</width>
    <height>812</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>99999</width>
    <height>99999</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Courier New</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>Генератор пакетов</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>8</y>
      <width>801</width>
      <height>741</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_Queue">
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Очередь пакетов</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTableWidget" name="tableWidget_PacketsQueue">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Courier New</family>
         <pointsize>8</pointsize>
        </font>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="tabKeyNavigation">
        <bool>false</bool>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <column>
        <property name="text">
         <string>Тип пакета</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Адрес отправителя</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Адрес получателя</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Порт отправителя</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Порт получателя</string>
        </property>
       </column>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_BottomButtons">
       <item>
        <widget class="QPushButton" name="pushButton_sendAll">
         <property name="text">
          <string>Отправить все пакеты</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_addPacket">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Создать пакет</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>8</y>
      <width>581</width>
      <height>741</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_Type_Fields">
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Настройка пакета</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabWidget_setPacket">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_TCP">
        <attribute name="title">
         <string>TCP</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>561</width>
           <height>671</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_FieldsTCP">
          <item row="12" column="2">
           <widget class="QLineEdit" name="lineEdit_paddingTCP"/>
          </item>
          <item row="2" column="2">
           <widget class="QLineEdit" name="lineEdit_dstPortTCP"/>
          </item>
          <item row="12" column="3">
           <widget class="QCheckBox" name="checkBox_autoPaddingTCP">
            <property name="text">
             <string>Авто</string>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QLineEdit" name="lineEdit_reservedTCP"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Флаги</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>SEQ</string>
            </property>
           </widget>
          </item>
          <item row="9" column="3">
           <widget class="QCheckBox" name="checkBox_autoChecksumTCP">
            <property name="text">
             <string>Авто</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Смещение данных</string>
            </property>
           </widget>
          </item>
          <item row="13" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Данные</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
          <item row="6" column="3">
           <widget class="QCheckBox" name="checkBox_autoReservedTCP">
            <property name="text">
             <string>Авто</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="lineEdit_srcPortTCP"/>
          </item>
          <item row="3" column="2">
           <widget class="QLineEdit" name="lineEdit_seqTCP"/>
          </item>
          <item row="5" column="2">
           <widget class="QLineEdit" name="lineEdit_dataOffsetTCP"/>
          </item>
          <item row="15" column="2" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayout_2"/>
          </item>
          <item row="8" column="2">
           <widget class="QLineEdit" name="lineEdit_windowSizeTCP"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Порт отправки</string>
            </property>
           </widget>
          </item>
          <item row="14" column="0" colspan="4">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="pushButton_deletePacketTCP">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Удалить пакет</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_sendPacketTCP">
              <property name="text">
               <string>Отправить пакет</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_saveTCP">
              <property name="text">
               <string>Сохранить пакет</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Заполнение</string>
            </property>
           </widget>
          </item>
          <item row="11" column="2">
           <widget class="QPlainTextEdit" name="plainTextEdit_optionsTCP"/>
          </item>
          <item row="9" column="2">
           <widget class="QLineEdit" name="lineEdit_checksumTCP"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>ACK</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QLineEdit" name="lineEdit_ackTCP"/>
          </item>
          <item row="13" column="2">
           <widget class="QPlainTextEdit" name="plainTextEdit_dataTCP"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Зарезервированное поле</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Порт назначения</string>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Указатель на срочные данные</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Размер окна</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Контрольная сумма</string>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="label_13">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Опции</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
          <item row="5" column="3">
           <widget class="QCheckBox" name="checkBox_autoDataOffsetTCP">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Авто</string>
            </property>
           </widget>
          </item>
          <item row="10" column="2">
           <widget class="QLineEdit" name="lineEdit_urgentPointerTCP"/>
          </item>
          <item row="7" column="2">
           <layout class="QGridLayout" name="gridLayout_Flags">
            <item row="3" column="1">
             <widget class="QCheckBox" name="checkBox_finTCP">
              <property name="text">
               <string>[5] FIN</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QCheckBox" name="checkBox_pshTCP">
              <property name="text">
               <string>[2] PSH</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QCheckBox" name="checkBox_urgTCP">
              <property name="text">
               <string>[0] URG</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QCheckBox" name="checkBox_ackTCP">
              <property name="text">
               <string>[1] ACK</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QCheckBox" name="checkBox_rstTCP">
              <property name="text">
               <string>[3] RST</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="checkBox_synTCP">
              <property name="text">
               <string>[4] SYN</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>dec</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>dec</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>dec</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>dec</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>bin</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>dec</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>hex</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>hex</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_UDP">
        <attribute name="title">
         <string>UDP</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tab_ICMP">
        <attribute name="title">
         <string>ICMP</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tab_IP">
        <attribute name="title">
         <string>IP</string>
        </attribute>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1412</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Инструменты</string>
    </property>
    <addaction name="actionSetInterface"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionSetInterface">
   <property name="text">
    <string>Настройка интерфейсов...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
